parquet-column/src/main/java/org/apache/parquet/column/values/bytestreamsplit/ByteStreamSplitValuesReader.java

```diff
@@ -58,15 +58,19 @@ protected void gatherElementDataFromStreams(byte[] gatheredData)
   public void initFromPage(int valuesCount, ByteBufferInputStream stream) throws ParquetDecodingException, IOException {
```

Comment:
It seems the input value valueCount is not used anymore

Reply:
Added a check to use `valuesCount` as upper bound.